### PR TITLE
New config option: use_crc

### PR DESCRIPTION
**Pull request by mgkahn**
The current blob.download_to_file() method does not do CRC32 integrity checking when downloading files to local disk. File corruption during network transfer will not be detected.

This PR adds new (optional) tag: use_crc. 

- If use_crc=0 (default), no CRC32 integrity checking will be done.  This is the current functionality
- If use_crc=1, CRC32 integrity checking will be done on each downloaded file. Requires google-crc32c module.

Tested on Python 9 and Python 11.

### CODE REVIEW

File: config_example.yaml [hunk 1]
```diff
@@ -40,6 +40,11 @@ spark_bq_jar: gs://spark-lib/bigquery/spark-bigquery-with-dependencies_2.13-0.31
 # if set to 0, it will leave the merged csv files for CDM tables in GCS, default is 1 i.e. csv files will be deleted
 delete_merged_csvs_from_bucket: 0
 
+# if set to 0, do not use CRC validation on export. Any non-zero number, use crc validation on export. Default = 0 (no CRC validation)
+# Requires google-crc32c module installed
+use_crc: 1
```
Comment (smitalimaye) [inline config_example.yaml]: Should this be set to default? i.e. 0?

Comment (mgkahn) [reply config_example.yaml]: @smitalimaye Ugh, yes. I left in the value that I am using at Colorado. I don't know how to modify the PR. Can you make this change on your side?

**Review by smitalimaye — APPROVED** [review]

**Comment by smitalimaye** [comment]
@mgkahn Feel free to merge the PR whenever you are ready

**Comment by mgkahn** [comment]
@smitalimaye : I do not think I have sufficient access to perform the final merge. The "Merge Pull Request" button does not appear here even though I am logged into GitHub.
BTW: The current code exits with an exception if I get a CRC error while downloading a file. I am working on extending the current code to allow for up to three retries before moving on to the attempting the next file in the d/l list. My intent is to count the number of files in the local directory following PySpark completion (or maybe grepping the log file) in my calling BASH script before deciding to move on to ZIP or stop processing and manually d/l the missing file(s) from the console. 
Is this acceptable or would you prefer that processing stop (not move on to the next file) if all retries on the current file fail? I can make this expansion a separate PR when it is done or withdraw this PR and submit this expanded version.

**Comment by smitalimaye** [comment]
> extending the current code to allow for up to three retries before moving on to the attempting the next file

Sure, extending the current code to allow for up to three retries before moving on to the attempting the next file sounds good. I can merge this PR now.

**Pull request merged**